homeassistant/components/hunterdouglas_powerview/button.py

```diff
@@ -0,0 +1,142 @@
+"""Buttons for Hunter Douglas Powerview advanced features."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any, Final
+
+from aiopvapi.resources.shade import BaseShade, factory as PvShade
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    COORDINATOR,
+    DEVICE_INFO,
+    DOMAIN,
+    PV_API,
+    PV_ROOM_DATA,
+    PV_SHADE_DATA,
+    ROOM_ID_IN_SHADE,
+    ROOM_NAME_UNICODE,
+)
+from .coordinator import PowerviewShadeUpdateCoordinator
+from .entity import ShadeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class PowerviewButtonDescriptionMixin:
+    """Mixin to describe a Button entity."""
+
+    press_action: Callable
```

Comment:
Please type the Callable https://docs.python.org/3/library/typing.html#typing.Callable

Reply:
Done too